servers/file_system/fat/detect_fat.c

```diff
@@ -98,15 +118,36 @@ bool detect_fat(fat_info_type *fat_info)
         message_parameter.data = &ipc_block_read;
         message_parameter.length = sizeof(ipc_block_read_type);
 
-        ipc_block_read.start_block_number = 1;
+        ipc_block_read.start_block_number = bios_parameter_block->reserved_sectors;
         ipc_block_read.number_of_blocks = bios_parameter_block->fat_size_16;
 
-        ipc_send(fat_info->block_structure.output_mailbox_id, &message_parameter);
+        if (ipc_send(fat_info->block_structure.output_mailbox_id, &message_parameter) != IPC_RETURN_SUCCESS)
+        {
+            log_print_formatted(
+                &log_structure,
+                LOG_URGENCY_ERROR,
+                "ipc_send to mailbox %u failed (from %s:%u)",
+                fat_info->block_structure.output_mailbox_id,
+                __FILE__, __LINE__
+            );
+            return FALSE;
+        }
 
-        memory_set_uint8_t((uint8_t *) &global_fat, 0, 16384);
+        memory_set_uint32_t((uint32_t *) &global_fat, 0xDEADBEEF, 16384 / sizeof(uint32_t));
```

Comment:
Rationale: Initializing the memory with non-zero data can simplify debugging sometimes, when inspecting memory in debuggers.